src/widget/form/settings/avform.cpp

```diff
@@ -227,12 +227,14 @@ void AVForm::updateVideoModes(int curIndex)
     QSize prefRes = Settings::getInstance().getCamVideoRes();
     unsigned short prefFPS = Settings::getInstance().getCamVideoFPS();
     // Iterate backwards to show higest resolution first.
+    videoModes.clear();
     for(auto iter = bestModeInds.rbegin(); iter != bestModeInds.rend(); ++iter)
     {
         int i = iter->second;
-        VideoMode mode = videoModes[i];
+        VideoMode mode = allVideoModes[i];
+        videoModes.push_back(mode);
```

Comment:
I wondered for a second, grepped the source and found it was more or less random apparently, so I kept what I had, but I'll change it.

```console
$ git grep -F .append | wc -l
42
$ git grep -F .push_back | wc -l
33
```